pkg/asset/installconfig/ovirt/datacenter.go

```diff
@@ -0,0 +1,33 @@
+package ovirt
+
+import (
+	ovirtsdk4 "github.com/ovirt/go-ovirt"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type dataCenterConfig struct {
+	conn         *ovirtsdk4.Connection // ovirtsdk4 connection data
+	searchFilter string                // A search filter for datacenters
+}
+
+// datacentersAvailable look for all datacenters available in the system.
+// Users can provide the filter for the search. Example: status=down
+// If search filter not provided, the default filter will be "status=up"
+// Returns type: *ovirtsdk.DataCentersServiceListResponse
+func datacentersAvailable(d dataCenterConfig) (*ovirtsdk4.DataCentersServiceListResponse, error) {
```

Comment:
nit: my preference would be to not use argument structures because they make reading the input api a little difficult due to indirection.

```suggestion
func datacentersAvailable(conn *ovirtsdk4.Connection, searchFilter string) (*ovirtsdk4.DataCentersServiceListResponse, error) {
```